### PR TITLE
Remove outdated fields

### PR DESCRIPTION
**Pull request by yzdann**
### All Submissions:

* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?
* [ ] Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if there is one).
* [ ] Update docs/release-notes/release-notes.md if your changes should be included in the release notes for the next release.

Related: #606

### CODE REVIEW

**Comment by yzdann** [comment]
@fabritsius

**Comment by fabritsius** [comment]
Hey @yzdann,

Thanks you for this PR. You are correct that not all of the fields should be sent during PATCH request.
There is another PR #615 where I try to remove the need of such filtering entirely and make it so that these fields won't be mixed in the object in the first place.

It is possible that some filtering will be required even after my changes so leaving this PR as open for now.

File: pkg/controller/atlasdeployment/advanced_deployment.go [hunk 1]
```diff
@@ -124,6 +163,9 @@ func MergedAdvancedDeployment(advancedDeployment mongodbatlas.AdvancedCluster, s
 
 // AdvancedDeploymentsEqual compares two Atlas Advanced Deployments
 func AdvancedDeploymentsEqual(log *zap.SugaredLogger, deploymentAtlas mongodbatlas.AdvancedCluster, deploymentOperator mongodbatlas.AdvancedCluster) bool {
```
Comment (sunchill06) [inline pkg/controller/atlasdeployment/advanced_deployment.go:165]: Would be good to have a test case for this?

**Comment by yzdann** [comment]
Resolve https://github.com/mongodb/mongodb-atlas-kubernetes/issues/648

**Comment by yzdann** [comment]
covered by https://github.com/mongodb/mongodb-atlas-kubernetes/pull/672

**Pull request closed without merging**